CONTRIBUTING.md

```diff
@@ -178,38 +178,60 @@ The main development branch is master.
-- before submitting a pull request, run `git rebase` on top of the latest `master`
+The latest state of development is on `master`, which must never fail `make test`. _Never_ force push `master`, unless fixing broken git history (which we rarely do anyways). 
+
+To begin contributing, create a development branch either on github.com/tendermint/tendermint, or your fork (using `git remote add origin`). 
+
+Make changes, and before submitting a pull request, update the `CHANGELOG_PENDING.md` to record your change. Also, run either `git rebase` or `git merge` on top of the latest `master`. (Since pull requests are squash-merged, either is fine!)
+
+Once you have submitted a pull request label the pull request with either `R:minor`, if the change should be included in the next minor release, or `R:major`, if the change is meant for a major release.
+
+Sometimes (often!) pull requests get out-of-date with master, as other people merge different pull requests to master. It is our convention that pull request authors are responsible for updating their branches with master. (This also means that you shouldn't update someone else's branch for them; even if it seems like you're doing them a favor, you may be interfering with their git flow in some way!) 
+
+#### Merging Pull Requests 
+
+It is also our convention that authors merge their own pull requests, when possible. External contributors may not have the necessary permissions to do this, in which case, a member of the core team will merge the pull request once it's been approved. 
+
+Before merging a pull request: 
 
-When you have submitted a pull request label the pull request with either `R:minor`, if the change can be accepted in a minor release, or `R:major`, if the change is meant for a major release.
+- Ensure pull branch is up-to-date with a recent `master` (GitHub won't let you merge without this!) 
+- Run `make test` to ensure that all tests pass
+- [Squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) merge pull request
 
-### Pull Merge Procedure
+#### Pull Requests for Minor Releases 
 
-- ensure pull branch is based on a recent `master`
-- run `make test` to ensure that all tests pass
-- [squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) merge pull request
-- the `unstable` branch may be used to aggregate pull merges before fixing tests
+If your change should be included in a minor release, please also open a PR against the long-lived minor release candidate branch (e.g., `rc1/v0.33.5`) _immediately after your change has been merged to master_.
+
+You can do this by cherry-picking your commit off master:
+
+```
+$ git checkout rc1/v0.33.5
+$ git checkout -b {new branch name}
+$ git cherry-pick {commit SHA from master} 
+# may need to fix conflicts, and then use git add and git cherry-pick --continue
+$ git push origin {new branch name}
```

Comment:
should we add here to add the changelog entry to `CHANGELOG.md` instead of the pending? would reduce the steps below for the minor release

Reply:
oh, that's interesting. we should just have a running changelog, of course! I'll update all this.

Reply:
actually, I changed my mind on this. it's probably easier for all authors to just cherry pick their commits and then for the person doing the release to copy changelog_pending over all at once, rather than asking all authors to add an extra commit to change their changelog entries. I liked the idea though :)